### DISCUSSION TITLE
Opencart 4 Guest Checkout

### DISCUSSION THREAD

**Discussion by testricha124** [opencart/opencart#12676] [Q&A]
Hello, Community
I checked and find bugs, in opencart 4 latest demo at (demo.opencart.com) guest checkout is not working ?
I am trying to add product in to cart without customer login is not added product in cart and not getting any appropriate message. after customer login add to cart functionality working well ..

So you removed guest checkout functionality from oc 4.x. ??

**Comment by mhcwebdesign** [comment]
Try the latest master branch, or version 4.0.2.2 where this works just fine.
